stdlib/public/core/Sort.swift.gyb

```diff
@@ -364,9 +364,272 @@ public func sorted<
 ) -> [C.Generator.Element] {
   fatalError("unavailable function can't be called")
 }
+
+
+/// Performs a binary search in `from`..<`to` for the value at `val`,
+/// preserving the lower ordering of `val` in case of equal matches.
+///
+/// - Requires: The values in `from`..<`to` must already be in order.
+/// - Returns: The stable insertion point for the value in `val`.
+private func _stableSearchWithLower<
```

Comment:
Please don't use `private`, it is subtly broken in the standard library -- use `internal` instead.

I am not sure what does 'lower ordering' mean by the way.